src/cogent3/app/composable.py

```diff
@@ -129,651 +129,6 @@ def to_json(self):
-                input_md5 = getattr(m, "md5", None)
-                logger.log_message(input_id, label="input")
-                if input_md5:
-                    logger.log_message(input_md5, label="input md5sum")
-
-                if isinstance(outcome, NotCompleted):
-                    # log error/fail details
-                    logger.log_message(
-                        f"{outcome.origin} : {outcome.message}", label=outcome.type
-                    )
-                    continue
-                elif not outcome:
-                    # other cases where outcome is Falsy (e.g. None)
-                    logger.log_message(f"unexpected value {outcome!r}", label="FAIL")
-                    continue
-
-                logger.log_message(outcome, label="output")
-                logger.log_message(outcome.md5, label="output md5sum")
-
-        if am_writer:
-            finish = time.time()
-            taken = finish - start
-
-            logger.log_message(f"{taken}", label="TIME TAKEN")
-            logger.shutdown()
-            self.data_store.add_file(log_file_path, cleanup=cleanup, keep_suffix=True)
-            self.data_store.close()
-
-        # now reconnect input
-        if process is not self:
-            self = process + self
-
-        return results
-
-
-class ComposableTabular(Composable):  # pragma: no cover
-    def __init__(self, **kwargs):
-        super(ComposableTabular, self).__init__(**kwargs)
-        discontinued(
-            "class",
-            "ComposableTabular",
-            "2023.1",
-            "see developer docs for new class hierarchy",
-        )
-
-
-class ComposableSeq(Composable):  # pragma: no cover
-    def __init__(self, **kwargs):
-        super(ComposableSeq, self).__init__(**kwargs)
-        discontinued(
-            "class",
-            "ComposableSeq",
-            "2023.1",
-            "see developer docs for new class hierarchy",
-        )
-
-
-class ComposableAligned(Composable):  # pragma: no cover
-    def __init__(self, **kwargs):
-        super(ComposableAligned, self).__init__(**kwargs)
-        discontinued(
-            "class",
-            "ComposableAligned",
-            "2023.1",
-            "see developer docs for new class hierarchy",
-        )
-
-
-class ComposableTree(Composable):  # pragma: no cover
-    def __init__(self, **kwargs):
-        super(ComposableTree, self).__init__(**kwargs)
-        discontinued(
-            "class",
-            "ComposableTree",
-            "2023.1",
-            "see developer docs for new class hierarchy",
-        )
-
-
-class ComposableModel(Composable):  # pragma: no cover
-    def __init__(self, **kwargs):
-        super(ComposableModel, self).__init__(**kwargs)
-        discontinued(
-            "class",
-            "ComposableModel",
-            "2023.1",
-            "see developer docs for new class hierarchy",
-        )
-
-
-class ComposableHypothesis(Composable):  # pragma: no cover
```

Comment:
Did we replace all these classes in new decorator model? ComposableHypothesis, ComposableDistance and others. I am not sure about this.

Reply:
Absolutely! The `define_app` decorator completely removed the need for subclassing.